### PR TITLE
Grafana agent

### PR DESCRIPTION
**Pull request by 123MwanjeMike**
# Add k8s grafana agent installation files

**_WHAT DOES THIS PR DO?_**
- Adds k8s grafana agent installation files

**_WHAT JIRA STORY/TASK IS RELATED TO THIS PR?_**
- [PLAT-1049](https://airqoteam.atlassian.net/browse/PLAT-1049)

**_HOW DO I TEST OUT THIS PR?_**

**Assumption:** The grafana-agent and grafana-agent-logs [config maps](https://drive.google.com/drive/folders/10Aa2t4dtWErc-odA-_jZL0Aio-nC8d6h?usp=sharing) have already been created in the cluster(staging) you are testing with
- Run `kubectl apply -f k8s/grafana/agent/statefulSet.yaml` to install the agent statefulSet
- Run the `k8s/grafana/deploy-kube-state-metrics.sh` shell script to deploy the kube-state-metrics
- Run `kubectl apply -f k8s/grafana/agent/daemonSet.yaml` to install the agent daemonSet

**_WHICH ENDPOINTS SHOULD BE READY FOR TESTING?:_** 
[Kubernetes Integration](https://airqo.grafana.net/d/lAoEVhD7z/home-kubernetes-integration?orgId=1&refresh=30s&search=open)

**_IS THERE ANY JENKINS CONSOLE LINK FOR CODE COVERAGE AND BUILD INFO?_**
- N/A

**_ARE THERE ANY RELATED PRs?_**
- N/A

### CODE REVIEW

**Comment by NoahNsimbe** [comment]
Hi @123MwanjeMike could you share a link to the sample config maps

**Comment by 123MwanjeMike** [comment]
> Hi @123MwanjeMike could you share a link to the sample config maps

Sure thing @NoahNsimbe 
I've updated the description with the link to the config maps

**Review by NoahNsimbe — APPROVED** [review]
LGTM. Thanks @123MwanjeMike

**Review by Baalmart — COMMENTED** [review]
@123MwanjeMike , please add some review comments to your changed files to guide the reviewers. Thanks!

File: k8s/grafana/deploy-kube-state-metrics.sh [hunk 1]
```diff
@@ -0,0 +1,4 @@
+#!/bin/sh
+helm repo add prometheus-community https://prometheus-community.github.io/helm-charts
```
Comment (123MwanjeMike) [inline k8s/grafana/deploy-kube-state-metrics.sh:2]: Installation of kube-state-metrics via bash script

File: k8s/grafana/agent/daemonSet.yaml [hunk 2]
```diff
@@ -0,0 +1,101 @@
+apiVersion: v1
```
Comment (123MwanjeMike) [inline k8s/grafana/agent/daemonSet.yaml:1]: daemonset service account

File: k8s/grafana/agent/daemonSet.yaml [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+  - watch
+- nonResourceURLs:
+  - /metrics
+  verbs:
+  - get
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: grafana-agent-logs
```
Comment (123MwanjeMike) [inline k8s/grafana/agent/daemonSet.yaml:33]: grafana-agent-logs clusterrolebinding

File: k8s/grafana/agent/daemonSet.yaml [hunk 4]
```diff
@@ -0,0 +1,101 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: grafana-agent-logs
+  namespace: monitoring
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
```
Comment (123MwanjeMike) [inline k8s/grafana/agent/daemonSet.yaml:8]: grafana agent logs clusterrole

File: k8s/grafana/agent/statefulSet.yaml [hunk 5]
```diff
@@ -0,0 +1,108 @@
+apiVersion: v1
+kind: ServiceAccount
+metadata:
```
Comment (123MwanjeMike) [inline k8s/grafana/agent/statefulSet.yaml:3]: grafana agent service account

File: k8s/grafana/agent/statefulSet.yaml [hunk 6]
```diff
@@ -0,0 +1,108 @@
...
+          name: http-metrics
+        volumeMounts:
+        - mountPath: /var/lib/agent
+          name: agent-wal
+        - mountPath: /etc/agent
+          name: grafana-agent
+      serviceAccount: grafana-agent
+      volumes:
+      - configMap:
+          name: grafana-agent
```
Comment (123MwanjeMike) [inline k8s/grafana/agent/statefulSet.yaml:101]: grafana agent config map

**Comment by 123MwanjeMike** [comment]
> @123MwanjeMike , please add some review comments to your changed files to guide the reviewers. Thanks!

Thanks @Baalmart 
Right awaya

**Review by danielogen — APPROVED** [review]
LGTM

**Pull request merged**